docs/content/direct/hub-chart.md

```diff
@@ -0,0 +1,262 @@
+```
+
+The above command will install KubeFlex and the Post Create Hooks, but no Control Planes.
+Please remeber to add `--set "kubeflex-operator.isOpenShift=true"`, when installing into an OpenShift cluster.
+
+User defined control planes can be added using additional value files of `--set` arguments, _e.g._:
+
+- add a single ITS named its1 of default vcluster type: `--set-json='ITSes=[{"name":"its1"}]'`
+- add two ITSes named its1 and its2 of of type vlcuster and host, respectevely: `--set-json='ITSes=[{"name":"its1"},{"name":"its2","type":"host"}]'`
+- add a single WDS named wds1 of default k8s type connected to the one and only ITS: `--set-json='WDSes=[{"name":"wds1"}]'`
+
+A KubeStellar Hub installation compatible with the common setup suitable for Common Setup described in the [examples](./examples.md) could be achieved with the following command:
+
+```shell
+helm upgrade --install ks-hub oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set-json='ITSes=[{"name":"its1"}]' \
+  --set-json='WDSes=[{"name":"wds1"}]'
+```
+
+After the initial instllation is completed, there are two main ways to install additional control planes (_e.g._, create a second `wds2` WDS):
+
+1. Upgrade the initial chart. This choice requires to relist the existing control planes, which would otherwise be deleted:
+
+```shell
+helm upgrade --install ks-hub oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set-json='ITSes=[{"name":"its1"}]' \
+  --set-json='WDSes=[{"name":"wds1"},{"name":"wds2"}]'
+```
+
+2. Install a new chart with a different name. This choice does not requires to relist the existing control planes, but requies to disable the reinstallation of KubeFlex and PCHs:
+
+```shell
+helm upgrade --install add-wds2 oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set='kubeflex-operator.install=false,InstallPCHs=false' \
+  --set-json='WDSes=[{name":"wds2"}]'
+```
+
+## Obtaining the Control Planes kubeconfigs
+
+The following code can be used to wait for the Ready state of all the KubeFlex Control Planes in the hosting cluster:
+
+```shell
+echo "Waiting for all Control Planes to be ready..."
+for cpname in `kubectl get controlplane -o name`; do
+  cpname=${cpname##*/}
+  while [[ $(kubectl get cp $cpname -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]]; do
+    echo "Waiting for Control Plane $cpname..."
+    sleep 5
+  done
+  echo $cpname
+done
+```
+
+The kubeconfig of the KubeFlex control planes can be obtained both with and without the use of `kflex` CLI.
+
+1. using `kflex` CLI
+
+The kubeconfig of a `$cpname` Control Plane can be added as a context of the existing kubeconfig by using the command:
+
+```shell
+kflex ctx $cpname
+```
+
+Note that the command above will also switch the current context to `$cpname`, use `kflex ctx` to switch back to the initial context.

```

Comment:
Remember that this does not work if that context was renamed after kflex created it.

Reply:
ok but in this workflow we do not use `kflex init`

Reply:
Hmm, now I am wondering now the user's kubeconfig file gets what `kflex ctx` needs.

Reply:
I'm not sure, it may require looking into the go code

Reply:
You might find https://github.com/kubestellar/kubeflex/issues/202 helpful.

Reply:
@MikeSpreitzer good point. I have proposed a solution in https://github.com/kubestellar/kubestellar/issues/2091#issuecomment-2092951213

Reply:
It seems to me that we have a fundamental conflict here between trying to do everything with a Helm chart (which can only affect the contents of a Kubernetes cluster) and KubeFlex keeping important state on the user's machine (which is inaccessible from the Kubernetes cluster).

If we are going to solve this by issuing one `kflex` command, why make it obscure --- why not just make it `kflex init`?

Reply:
@MikeSpreitzer this was my initial intention when I created the original kflex PR which was voted down. Besides some nomenclature https://github.com/kubestellar/kubeflex/pull/222
In any case I don't think this can be address in this PR.

Reply:
I think that this has to be addressed in this PR. Without the needed extension in the user's kubeconfig file, this will not work at all.

Perhaps the earlier discussion was not sufficiently informed by this difficulty. I do not remember thinking about this at that time.

Reply:
I have realized that my previous statement is not quite right; https://github.com/kubestellar/kubeflex/pull/254 is a relevant doc update about KubeFlex and how it remembers the name of the kubeconfig context used to access the hosting cluster. The key realization is that `kflex ctx $cpname` _will_ work even if the hosting cluster context name is not yet stored in the user's kubeconfig **provided** that the current context is the one for accessing the hosting cluster. So I think that we can proceed in either of two ways.

1. Keep the chart's initial condition at "hosting cluster exists and is accessed by the kubeconfig current-context" and forbidding the user to change the kubeconfig current-context by any other means between (a) instantiating the chart and (b) using `kflex ctx $cpname` or `kflex create`.
2. Chang the chart's initial condition to "the hosting cluster has been kflex-initted (either by `kflex init` or instantiating a Helm chart) AND kflex has stored the name of the kubeconfig context used to access the hosting cluster" --- which will be less fragile if we upgrade `kflex` to store that more eagerly (https://github.com/kubestellar/kubeflex/issues/255).

Reply:
After some discussion we decided to continue with the current chart design. To help with the sharp edge of `kflex ctx` being unable to do its job because the kubeconfig extension is missing, we can make the error message clearly explain the situation and how to remedy it. BTW, https://github.com/kubestellar/kubeflex/issues/255 suggests related improvements to `kflex`.